LuaSkin/LuaSkin/Skin.m

```diff
@@ -335,7 +335,24 @@ - (void)checkArgs:(int)firstArg, ... {
                 lsType = LS_TFUNCTION;
                 break;
             case LUA_TTABLE:
-                lsType = LS_TTABLE;
+                if (spec & LS_TTYPEDTABLE) {
+                    lsType = LS_TTYPEDTABLE;
+                    char *expectedTableTag = va_arg(args, char*);
+// NOTE: Make sure LS_TTYPEMASK in Skin.h is updated as options are added. Since va_args can't check type for us, we have to do this to make sure we get an address rather than an integer cast as an address... it's a bit of a hack for bit-position based flags since at some point we enter the realm of a valid setting being a valid address, but its the best I've been able to come up with so far...
+                    if (((size_t)expectedTableTag & ~LS_TTYPEMASK) == 0)
```

Comment:
@asmagill I think I'm fine with crashing to motivate the developer, that's always been my approach to `LS_TUSERDATA` :)